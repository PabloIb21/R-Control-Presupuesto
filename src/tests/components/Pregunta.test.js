import React from 'react';
import { mount } from 'enzyme';
import Pregunta from '../../components/Pregunta';

describe('<Pregunta />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Pregunta
        guardarPresupuesto={() => {}}
        guardarRestante={() => {}}
        actualizarPregunta={() => {}}
      />
    );
  });

  test('mount', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should display error when the form is sent empty or with a negative number', () => {
    wrapper.find('form').simulate('submit');

    expect(wrapper.find('.error').exists()).toBe(true);
    expect(wrapper.find('.error').text()).toBe('El presupuesto es incorrecto');
  });

  test('should display no error when the form is sent correctly', () => {
    wrapper.find('input').at(0).simulate('change', { target: { value: '1000' } });
    wrapper.find('form').simulate('submit');

    expect(wrapper.find('.error').exists()).toBe(false);
  });
});