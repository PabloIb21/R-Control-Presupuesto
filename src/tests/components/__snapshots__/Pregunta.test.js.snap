// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pregunta /> mount 1`] = `
Array [
  <h2>
    Coloca tu presupuesto
  </h2>,
  <form
    onSubmit={[Function]}
  >
    <input
      className="u-full-width"
      onChange={[Function]}
      placeholder="Coloca tu presupuesto"
      type="number"
    />
    <input
      className="button-primary u-full-width"
      type="submit"
      value="Definir presupuesto"
    />
  </form>,
]
`;
